File: frontend/js/services/movie-service.js
import { API, ApiClient } from '../config/api.js';

export const MovieService = {
  async getAll() {
    return ApiClient.get(API.MOVIES.ALL, false);
  },

  async getById(id) {
    return ApiClient.get(API.MOVIES.BY_ID(id), false);
  },

  async getByGenre(genre) {
    return ApiClient.get(API.MOVIES.BY_GENRE(genre), false);
  },

  async search(query) {
    return ApiClient.get(API.MOVIES.SEARCH(encodeURIComponent(query)), false);
  },
};

File: movie-service/src/main/java/com/bookmyshow/movie/service/MovieService.java
package com.bookmyshow.movie.service;

import com.bookmyshow.movie.dto.MovieRequest;
import com.bookmyshow.movie.dto.MovieResponse;
import java.util.List;

public interface MovieService {
    MovieResponse create(MovieRequest req);
    MovieResponse getById(Long id);
    List<MovieResponse> getAll();
    List<MovieResponse> getFeatured();
    List<MovieResponse> getTrending();
    List<MovieResponse> getUpcoming();
    List<MovieResponse> getPopular();
    List<MovieResponse> getByGenre(String genre);
    List<MovieResponse> getByLanguage(String language);
    List<MovieResponse> search(String query);
}
